src/components/AddToCartModal/AddToCartModal.tsx

```diff
@@ -0,0 +1,76 @@
+import Modal from 'react-modal';
+import Image from 'next/image';
+import { Product } from '~/types/product';
+import Checkmark from '~/../public/authenticated/checkmark.svg';
+import Button from '~/components/Button';
+import closeModal from '~/../public/authenticated/close-modal.svg';
+
+export default function AddToCartModal({
+  openModal,
+  product,
+  onModalClose,
+}: {
+  openModal: boolean;
+  product: Product | null;
+  onModalClose: () => void;
+}) {
+  const modalStyles = {
+    content: {
+      margin: 'auto',
+      width: 'fit-content',
+      height: 'fit-content',
+      top: 32,
+      padding: 0,
+      display: 'inline-block',
+    },
+  };
```

Comment:
You can create this variable outside of the component definition, so it doesn't get created every time the component updates